app/src/main/java/com/amaze/filemanager/activities/superclasses/PermissionsActivity.java

```diff
@@ -102,4 +102,11 @@ private boolean isGranted(int[] grantResults) {
         void onPermissionGranted();
     }
 
+    @Override
+    protected void onDestroy() {
+        super.onDestroy();
+        for (int i = 0; i < permissionCallbacks.length; i++) {
+            permissionCallbacks[i] = null; // clear to let GC do its work
```

Comment:
This is redundant, the GC will clear automatically when necessary.

Reply:
From `java.util.Arraylist.remove` source code . 
``` java
    /**
     * Removes the element at the specified position in this list.
     * Shifts any subsequent elements to the left (subtracts one from their
     * indices).
     *
     * @param index the index of the element to be removed
     * @return the element that was removed from the list
     * @throws IndexOutOfBoundsException {@inheritDoc}
     */
    public E remove(int index) {
        if (index >= size)
            throw new IndexOutOfBoundsException(outOfBoundsMsg(index));

        modCount++;
        E oldValue = (E) elementData[index];

        int numMoved = size - index - 1;
        if (numMoved > 0)
            System.arraycopy(elementData, index+1, elementData, index,
                             numMoved);
        elementData[--size] = null; // clear to let GC do its work

        return oldValue;
    }
```
I think `elementData[--size] = null; // clear to let GC do its work` statement is needed